scarb/src/compiler/plugin/proc_macro/compilation.rs

```diff
@@ -33,3 +38,46 @@ impl SharedLibraryProvider for Package {
             .into_child(lib_name)
     }
 }
+
+pub fn compile_unit(unit: CompilationUnit, ws: &Workspace<'_>) -> Result<()> {
+    let main_package = unit.components.first().unwrap().package.clone();
+    let cmd = CargoCommand {
+        current_dir: main_package.root().to_path_buf(),
+        target_dir: main_package
+            .target_path(ws.config())
+            .path_unchecked()
+            .to_path_buf(),
+    };
+    let mut cmd: Command = cmd.into();
+    {
+        let _ = trace_span!("compile_proc_macro").enter();
+        let status = cmd
+            .status()
+            .with_context(|| format!("Failed to execute {:?}", cmd))?;
+        ensure!(
+            status.success(),
+            "failed to compile procedural macro plugin: {:?}",
+            cmd
+        );
+    }
+
+    Ok(())
+}
+
+struct CargoCommand {
+    current_dir: Utf8PathBuf,
+    target_dir: Utf8PathBuf,
+}
+
+impl From<CargoCommand> for Command {
+    fn from(args: CargoCommand) -> Self {
+        let mut cmd = Command::new("cargo");
+        cmd.stdout(Stdio::inherit());
+        cmd.stderr(Stdio::inherit());
+        cmd.current_dir(args.current_dir);
+        cmd.args(["build", "--release"]);
+        cmd.arg("--target-dir");
+        cmd.arg(args.target_dir);
+        cmd
```

Comment:
consider using https://github.com/software-mansion/scarb/blob/fe47e6ab16acf742ae1a028e84ef79f573330632/scarb/src/process.rs#L90-L148 for smoother UX. You'd have to add a configuration option to deal with Cargo's stderr, but it should be fine imo